sklearn/metrics/_pairwise_distances_reduction/_base.pyx

```diff
@@ -116,11 +116,14 @@ cdef class PairwiseDistancesReduction64:
         by parallelizing computation on the outer loop on chunks of X
         and reduce them.
 
-        This strategy dispatches chunks of Y uniformly on threads.
-        Each thread processes all the chunks of X in turn. This strategy is
-        a sequence of embarrassingly parallel subtasks (the inner loop on Y
-        chunks) with intermediate datastructures synchronisation at each
-        iteration of the sequential outer loop on X chunks.
+        This strategy dispatches tasks statically on threads.
+        Each task processes exactly only one chunk of X,
+        computing and reducing distances matrices between
+        vectors of this chunk and vectors of all chunks of Y,
+        one chunk of Y at a time.
+
+        This strategy is a sequence of embarrassingly parallel tasks
+        with no intermediate data structures synchronization at all.
```

Comment:
I would rather not use the word "sequence" here because the reader might understand that there are "sequential" (e.g. "non-parallel") sections or more than one synchronization point. The outer loop at line 143 is the parallel loop so it's embarrassingly parallel. There are no sequential computational steps in this function call (beyond the init and finalize steps).

```suggestion
        This strategy is embarrassingly parallel with no intermediate data
        structures synchronization at all.
```